src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,98 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int NON_ADD_FUNCTION_INDEX = 1;
+    private static final int GROW_SIZE_INDEX = 1;
+    private transient Object[] elementData = new Object[10];
+    private int size = 0;
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            grow(size + GROW_SIZE_INDEX);
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        rangeCheck(index);
+        Object[] elementData;
+        if (size == (elementData = this.elementData).length) {
+            elementData = grow(size + GROW_SIZE_INDEX);
+        }
+        System.arraycopy(elementData, index, elementData,
+                index + GROW_SIZE_INDEX, size - index);
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        grow(list.size() + size);
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        rangeCheck(index + NON_ADD_FUNCTION_INDEX);
```

Comment:
we don't need this if we change method rangeCheck(), it's not informative
```suggestion
        rangeCheck(index);
```